File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) { 
        int [] answer = new int [commands.length];
        int index = 0;
        // for each문으로 2차원배열 commands의 값들을 배열 s에 담아 돌림
        for (int[] s : commands) {
            // 잘라낸 배열 array를 담기위한 배열 slice 선언
            int [] slice = new int [s[1] - s[0] + 1];
            // 정렬위한 배열 arraySort 선언
            //int [] arraySort = new int [s[1] - s[0] + 1];
            
            // for문으로 slice에 잘라낸 array 담기
            for (int j = 0; j < s[1] - s[0] + 1; j++) {
                slice[j] = array[s[0]-1+j];
            }
            
            // slice 오름차순 정렬
            Arrays.sort(slice);
            // answer에 slice 인덱스값 담기
            answer[index] = slice[s[2]-1];
            index++;
        }

        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.go
package main

import "sort"

func solution(array []int, commands [][]int) []int {
	// 결과값 담을 슬라이스 선언
	var result []int

	// for range문을 이용하여 commands의 값을 value에 담아 돌림
	for _, value := range commands {
		var sliceSort []int
		var slice []int
		// value[1]의 값이 array의 길이와 같을 경우 (같으면 slice의 범위를 벗어남)
		if value[1] == len(array) {
			// array를 잘라 slice에 담는다
			slice = array[value[0]-1:]
			// slice의 값을 sliceValue에 담아 돌림
			for _, sliceValue := range slice {
				// sliceSort에 sliceValue의 값 추가
				sliceSort = append(sliceSort, sliceValue)
			}
			// sliceSort 오름차순 정렬
			sort.Ints(sliceSort)
			// result에 sliceSort의 인덱스값 추가
			result = append(result, sliceSort[value[2]-1])
			// 같지 않다면
		} else {
			// array를 잘라 slice에 담는다
			slice = array[value[0]-1 : value[1]]
			// slice의 값을 sliceValue에 담아 돌림
			for _, sliceValue := range slice {
				// sliceSort에 sliceValue의 값 추가
				sliceSort = append(sliceSort, sliceValue)
			}
			// sliceSort 오름차순 정렬
			sort.Ints(sliceSort)
			// result에 sliceSort의 인덱스값 추가
			result = append(result, sliceSort[value[2]-1])
		}
	}
	return result
}
